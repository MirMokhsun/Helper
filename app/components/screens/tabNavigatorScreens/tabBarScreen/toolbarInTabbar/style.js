import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: {
      flex: 1,
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: '#F5F5F7',
    },
    leftEmptyView: {
      width: 56,
      height: 56,
      justifyContent: 'center',
      alignItems: 'center'
    },
    rightComponent: {
      width: 56,
      height: 56,
      justifyContent: 'center',
      alignItems: 'center'
    },
    flex1: {
      flex: 1,
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
    },
    searchInputContent: {
      flex: 1,
      flexDirection: 'row',
      borderWidth: 1,
      borderColor: "#5A6772",
      borderRadius: 20,
      marginRight: 15,
      marginLeft: 15,
      height: 40,
      backgroundColor: 'transparent',
    },
    textInput: {
      flex: 1,
      fontFamily: 'Roboto-Light',
      fontSize: 14,
      color: '#5A6772',
      paddingHorizontal: 15,
      textDecorationLine: 'none',
    },
    textinputWithTextAlightRight: {
      flex: 1,
      fontFamily: 'Roboto-Light',
      textAlign: 'right',
      fontSize: 14,
      color: '#5A6772',
      paddingHorizontal: 15,
      textDecorationLine: 'none',
    },
  });