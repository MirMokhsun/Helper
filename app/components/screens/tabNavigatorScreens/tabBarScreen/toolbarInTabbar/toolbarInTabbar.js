import React, { Component } from 'react';
import { View, Platform, TouchableOpacity, TextInput } from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Title } from './title/title';
import IconSearchGray from '../../../../commonComponents/svg/Icon_search_gray';
import CrossSVG from '../../../../commonComponents/svg/crossSVG';
import { localization } from '../../../../../common/localization';
import { styles } from './style';
import { setSearchInHelpText } from '../../helpScreen/redux/searchInHelpAction';
import Utils from "../../../../../api/utils/utils";

const { isIphoneX } = Utils;

const constainerStyle = Platform.select({
  ios: {
    minHeight: isIphoneX ? 96 : 76,
    maxHeight: isIphoneX ? 96 : 76,
    paddingTop: isIphoneX ? 40 : 20,
  },
  android: {
    minHeight: 56,
    maxHeight: 56,
  },
});

export class ToolbarInTabbar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isShowSearchInput: false,
      pageTitle: '',
      routeName: ''
    }
  }

  UNSAFE_componentWillReceiveProps(nextProps, nextState) {
    const { navigation: { state: { index, routes } } } = nextProps;
    const { params: { pageTitle }, routeName } = routes[index]
    this.setState({ pageTitle, routeName });
  }

  showSearchInput = () => {
    this.setState({ isShowSearchInput: true }, () => this.searchInput.focus());
  }

  closeSearch = () => {
    const { setSearchInHelpText } = this.props;
    this.setState({ isShowSearchInput: false });
    setSearchInHelpText('');
  }

  inputChangeText = (text) => {
    const { setSearchInHelpText } = this.props;
    setSearchInHelpText(text);
  }

  getRefOnsearch = (input) => {
    this.searchInput = input;
  }

  render() {
    const { language } = this.props;
    const currentLanguage = localization[language];
    const { isShowSearchInput, pageTitle, routeName } = this.state;
    return (
      <View style={[styles.container, constainerStyle]}>
        <View style={styles.leftEmptyView} />
        {routeName === 'Help' ?
          (
            <View style={styles.flex1}>
              {isShowSearchInput ?
                (
                  <View style={styles.searchInputContent}>
                    <TextInput
                      ref={this.getRefOnsearch}
                      style={[styles.textInput, language === 'ar' ? styles.textinputWithTextAlightRight : null]}
                      underlineColorAndroid='transparent'
                      selectionColor="#5A6772"
                      onChangeText={this.inputChangeText}
                      placeholderTextColor='#000756'
                      placeholder={currentLanguage["helpSearch"]}
                      testID="SearchInputCountryID"
                      accessibilityLabel="SearchInputCountryID"
                    />
                  </View>
                ) : <Title pageName={pageTitle} />}
              <View style={styles.rightComponent} >
                {isShowSearchInput ? (
                  <TouchableOpacity onPress={this.closeSearch} testID="ToolbarInTabbarCloseID" accessibilityLabel="ToolbarInTabbarCloseID">
                    <CrossSVG color='#000756' />
                  </TouchableOpacity>
                ) : (
                    <TouchableOpacity onPress={this.showSearchInput} testID="ToolbarInTabbarSearchID" accessibilityLabel="ToolbarInTabbarSearchID">
                      <IconSearchGray color='#000756' />
                    </TouchableOpacity>
                  )
                }
              </View>
            </View>
          ) : (
            <View style={styles.flex1}>
              <Title pageName={pageTitle} />
              <View style={styles.rightComponent} />
            </View>
          )
        }
      </View>
    );
  }
}

const mapStateToProps = state => ({
  language: state.currentLanguage.language,
});
const mapDispatchToProps = dispatch => bindActionCreators({
  setSearchInHelpText,
}, dispatch);
export default connect(mapStateToProps, mapDispatchToProps)(ToolbarInTabbar);