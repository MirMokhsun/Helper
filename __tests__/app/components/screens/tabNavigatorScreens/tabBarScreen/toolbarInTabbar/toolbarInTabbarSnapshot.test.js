import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { ToolbarInTabbar } from '../../../../../../../app/components/screens/tabNavigatorScreens/tabBarScreen/toolbarInTabbar/toolbarInTabbar';

it('create snapshot ToolbarInTabbar', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<ToolbarInTabbar
        language={"ru"}
    />);
    expect(toolbarSnapshot).toMatchSnapshot();
});

