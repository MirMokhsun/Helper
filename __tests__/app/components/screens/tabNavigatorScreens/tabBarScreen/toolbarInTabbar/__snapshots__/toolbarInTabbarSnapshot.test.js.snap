// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot ToolbarInTabbar 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5F5F7",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "maxHeight": 96,
        "minHeight": 96,
        "paddingTop": 40,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 56,
        "justifyContent": "center",
        "width": 56,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Title
      pageName=""
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 56,
          "justifyContent": "center",
          "width": 56,
        }
      }
    />
  </View>
</View>
`;
